fluentd/run.sh

```diff
@@ -14,29 +14,74 @@ docker_uses_journal() {
     grep -q "^OPTIONS='[^']*--log-driver=journald" /etc/sysconfig/docker
 }
 
-if [ "${USE_JOURNAL:-}" != false ] ; then
-    if [ -z "${JOURNAL_SOURCE:-}" ] ; then
-        if [ -d /var/log/journal ] ; then
-            export JOURNAL_SOURCE=/var/log/journal
-        else
-            export JOURNAL_SOURCE=/run/log/journal
+if [ "${USE_MUX_SERVICE:-}" = true ] ; then
+    # mux service implies mux
+    export USE_MUX=true
+fi
+
+if [ "${USE_MUX:-}" != "true" ] ; then
+    if [ "${USE_JOURNAL:-}" != false ] ; then
+        if [ -z "${JOURNAL_SOURCE:-}" ] ; then
+            if [ -d /var/log/journal ] ; then
+                export JOURNAL_SOURCE=/var/log/journal
+            else
+                export JOURNAL_SOURCE=/run/log/journal
+            fi
         fi
-    fi
-    if [ -z "${USE_JOURNAL:-}" ] ; then
-        if docker_uses_journal ; then
-            export USE_JOURNAL=true
-        else
-            export USE_JOURNAL=false
+        if [ -z "${USE_JOURNAL:-}" ] ; then
+            if docker_uses_journal ; then
+                export USE_JOURNAL=true
+            else
+                export USE_JOURNAL=false
+            fi
         fi
     fi
+else
+    # mux requires USE_JOURNAL=true so that the k8s meta plugin will look
+    # for CONTAINER_NAME instead of the kubernetes.var.log.containers.* tag
+    export USE_JOURNAL=true
 fi
 
 IPADDR4=`/usr/sbin/ip -4 addr show dev eth0 | grep inet | sed -e "s/[ \t]*inet \([0-9.]*\).*/\1/"`
 IPADDR6=`/usr/sbin/ip -6 addr show dev eth0 | grep inet6 | sed "s/[ \t]*inet6 \([a-f0-9:]*\).*/\1/"`
 export IPADDR4 IPADDR6
 
 CFG_DIR=/etc/fluent/configs.d
-ruby generate_throttle_configs.rb
+if [ "${USE_MUX:-}" = "true" ] ; then
+    # copy our standard mux configs to the openshift dir
+    cp $CFG_DIR/input-*-mux.conf $CFG_DIR/openshift
```

Comment:
ok - so all of the mux config should be configmaps, as separate files in the logging-mux configmap, mounted at fluent/configs.d/user